### PR TITLE
[flutter_tools] remove substantial mocking from version test

### PR DESCRIPTION
**Pull request by jonahwilliams**
Refactored checkFlutterVersionFreshness into 2 methods, leaving the second easier to test. Updated test cases and removed some that were testing the mocks.

Part of #71511.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
woops i broke it

File: packages/flutter_tools/lib/src/version.dart [hunk 1]
```diff
@@ -837,3 +784,122 @@ enum VersionCheckResult {
...
+      ),
+      Future<void>.delayed(pauseTime),
+    ]);
+  }
+}
+
+/// The amount of time we wait before pinging the server to check for the
+/// availability of a newer version of Flutter.
+@visibleForTesting
+const Duration checkAgeConsideredUpToDate = Duration(days: 3);
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/version.dart:845]: Can you remove `Version.checkAgeConsideredUpToDate`, `Version.versionAgeConsideredUpToDate` etc?  Having two copies to keep in sync seems unnecessary.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/version.dart:845]: I thought I did, my bad

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**